pkgs/tools/audio/mpd-notification/default.nix

```diff
@@ -0,0 +1,57 @@
+{ lib
+, stdenv
+, pkg-config
+, fetchFromGitHub
+, systemd
+, file
+, iniparser
+, ffmpeg
+, libnotify
+, libmpdclient
+, discount
+}:
+
+stdenv.mkDerivation rec {
+  pname = "mpd-notification";
+  version = "0.8.7";
+
+  src = fetchFromGitHub {
+    owner = "eworm-de";
+    repo = "mpd-notification";
+    rev = version;
+    hash = "sha256-lBvx2eYxFJUAxR1LrjWHZUeAo+WnQKmPYJVAJTeXqHY=";
+  };
+
+  nativeBuildInputs = [
+    pkg-config
+  ];
+
+  buildInputs = [
+    iniparser
+    libnotify
+    file
+    ffmpeg
+    libmpdclient
+    discount
+   ];
+
+   installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    mv mpd-notification $out/bin
+
+    mkdir -p $out/lib/systemd/user
+    cp systemd/mpd-notification.service $out/lib/systemd/user
```

Comment:
I just noticed that the systemd unit contains a hardcoded path that needs to be substituted.

Can you also rename the PR to `mpd-notification: init at 0.8.7`?

Other than that, this should be good to merge

Reply:
Hmm the original daemon seems a bit funky on nix. Might just have to remove it for now. 
```
# (C) 2011-2023 by Christian Hesse <mail@eworm.de>
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.

[Unit]
Description=MPD Notification
Requires=dbus.socket
PartOf=graphical-session.target
# Do not require any service here! We do rely on mpd OR network (for
# a remote mpd instance). So let the user care.
# We want to order after, though. This makes sure the resource is
# available on start and mpd-notification can cleanly disconnect on
# system shutdown.
After=mpd.service network.target network-online.target
# Order after notification daemons to make sure it is stopped before.
After=dunst.service xfce4-notifyd.service
ConditionUser=!@system

[Service]
Type=notify
Restart=on-failure
ExecStart=/usr/bin/mpd-notification

[Install]
WantedBy=default.target
```

Reply:
What is the issue? the path in `ExecStart=` can easily be patched with something like 
```nix
postPatch = ''
   substituteInPlace systemd/mpd-notification.service --replace /usr $out
 '';
```
The "unrelated" units in `After=` should not cause any issues -- this just means that if systemd starts these units, they will be ordered before `mpd-notification`. This does not declare a dependency.

Reply:
Ah ok, thank you very much, should be good now